standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1772,6 +1773,22 @@ public void dropTable(String dbname, String name, boolean deleteData,
     dropTable(getDefaultCatalog(conf), dbname, name, deleteData, ignoreUnknownTab, ifPurge);
   }
 
+  @Override
+  public void dropTable(Table tbl, boolean deleteData, boolean ignoreUnknownTbl, boolean ifPurge) throws TException {
+    EnvironmentContext context = null;
+    if (ifPurge) {
```

Comment:
How will we handle purge with soft delete?

Reply:
good catch, it's not handled yet, as we don't support purge in Cleaner, however, we could pass `purge` attribute via the queue tblproperties.

Reply:
fixed, see Cleaner